### PR TITLE
2144: update formik files values on supporting upload success

### LINKED ISSUE #2144: Supporting Files dissapear if going next and back
**Describe the bug**
Supporting files entries disappear if you go back to the files step

**Steps To Reproduce**
- Start a new submission
- Uploading a manuscript and supporting files
- Click next
- Click back
- The supporting files are not there anymore. refreshing fixes it

**Expected behavior**
The supporting files should still be there

**Screenshots**

**Device, Operating System and Browser**

### PR DESCRIPTION
**Pull request by erezmus**
#### Background

Fixes issue with supporting files disappearing when going back into files step

#### Any relevant tickets

Closes #2144 

#### How has this been tested?
**Reviewers** ensure that these test requirements are also reviewed.

Please indicate the need for this change to be tested. If not please justify, if so then at what levels. 

- [x ] Unit / Integration tests
- [ ] Browser tests
- [ ] End2end tests

### CODE REVIEW

**Review by petereast — APPROVED** [review]

**Review by Cooryd — APPROVED** [review]

File: packages/component-submission/client/pages/FilesStepPage.js [hunk 1]
```diff
@@ -78,6 +78,27 @@ export class FilesStepPageComponent extends React.Component {
       })
   }
 
+  onSupportingFileUpload = file =>
+    new Promise((resolve, reject) => {
+      this.props.setFieldValue('fileStatus', 'CHANGING')
+      return this.props
```
Comment (Cooryd) [inline packages/component-submission/client/pages/FilesStepPage.js]: do we need to return this?

Comment (erezmus) [reply packages/component-submission/client/pages/FilesStepPage.js]: isn't this how it was behaving before?

**Comment by giorgiosironi** [comment]
Have to integrate https://github.com/elifesciences/builder-base-formula/pull/245 to make the last commit work.

**Comment by diversemix** [comment]
@erezmus please merge ASAP :)

**Pull request merged**